pkg/hvs/controllers/flavortemplate_controller_test.go

```diff
@@ -0,0 +1,434 @@
+/*
+ * Copyright (C) 2020 Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+package controllers_test
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/intel-secl/intel-secl/v3/pkg/hvs/controllers"
+	"github.com/intel-secl/intel-secl/v3/pkg/hvs/domain/mocks"
+	hvsRoutes "github.com/intel-secl/intel-secl/v3/pkg/hvs/router"
+	consts "github.com/intel-secl/intel-secl/v3/pkg/lib/common/constants"
+	"github.com/intel-secl/intel-secl/v3/pkg/model/hvs"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("FlavorTemplateController", func() {
+	var router *mux.Router
+	var w *httptest.ResponseRecorder
+	var flavorTemplateStore *mocks.MockFlavorTemplateStore
+	var flavorTemplateController *controllers.FlavorTemplateController
+	BeforeEach(func() {
+		router = mux.NewRouter()
+		flavorTemplateStore = mocks.NewFakeFlavorTemplateStore()
+
+		flavorTemplateController = controllers.NewFlavorTemplateController(flavorTemplateStore,
+			"../domain/schema/common.schema.json", "../domain/schema/flavor-template.json")
+	})
+
+	// Specs for HTTP Post to "/flavor-template"
+	Describe("Post a new FlavorTemplate", func() {
+		Context("Provide a valid FlavorTemplate data", func() {
+			It("Should create a new Flavortemplate and get HTTP Status: 200", func() {
+				router.Handle("/flavor-template", hvsRoutes.ErrorHandler(hvsRoutes.JsonResponseHandler(flavorTemplateController.Create))).Methods("POST")
+				flavorTemplateJson := `{
+					"label": "default-uefi",
+					"condition": [
+						"//host_info/vendor='Linux'",
+						"//host_info/tpm_version='2.0'",
+						"//host_info/uefi_enabled='true'",
+						"//host_info/suefi_enabled='true'"
+					],
+					"flavor_parts": {
+						"PLATFORM": {
+							"meta": {
+								"tpm_version": "2.0",
+								"uefi_enabled": true,
+								"vendor": "Linux"
+							},
+							"pcr_rules": [
+								{
+									"pcr": {
+										"index": 0,
+										"bank": "SHA256"
+									},
+									"pcr_matches": true,
+									"eventlog_equals": {}
+								}
+							]
+						},
+						"OS": {
+							"meta": {
+								"tpm_version": "2.0",
+								"uefi_enabled": true,
+								"vendor": "Linux"
+							},
+							"pcr_rules": [
+								{
+									"pcr": {
+										"index": 7,
+										"bank": "SHA256"
+									},
+									"pcr_matches": true,
+									"eventlog_includes": [
+										"shim",
+										"db",
+										"kek",
+										"vmlinuz"
+									]
+								}
+							]
+						}
+					}
+				}`
+
+				req, err := http.NewRequest(
+					"POST",
+					"/flavor-template",
+					strings.NewReader(flavorTemplateJson),
+				)
+				req.Header.Set("Accept", consts.HTTPMediaTypeJson)
```

Comment:
Expect(err).NotTo(HaveOccurred()) must be before processing any req details

Reply:
Addressed.